pkg/ingester/client/timeseries.go

```diff
@@ -276,6 +276,10 @@ func ReuseSlice(ts []PreallocTimeseries) {
 
 // ReuseTimeseries puts the timeseries back into a sync.Pool for reuse.
 func ReuseTimeseries(ts *TimeSeries) {
+	for i := 0; i < len(ts.Labels); i++ {
+		ts.Labels[i].Name = ""
```

Comment:
Suggest `// Name and Value may point into a large gRPC buffer, so clear the reference to allow GC`

Reply:
I accepted your suggestion, thanks!